core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -473,7 +480,7 @@ protected Location onReplicaShard(BulkShardRequest request, IndexShard indexShar
                 throw new IllegalStateException("Unexpected index operation: " + item.request());
             }
         }
-        return location;
+        return new WriteResult<>(null, location);
```

Comment:
this null feels wrong. Can we have a ReplicaWriteResult or something else that can handle the replica case? Maybe have WriteResult not require a response and have a PrimaryWriteResult inherit from and require a response?